utils/generate-command-code.py

```diff
@@ -319,35 +333,30 @@ def _doc_flags_code():
                 s += "CMD_DOC_%s|" % flag
             return s[:-1] if s else "CMD_DOC_NONE"
 
-        def _key_specs_code():
-            s = ""
-            for spec in self.key_specs:
-                s += "{%s}," % KeySpec(spec).struct_code()
-            return s[:-1]
-
-        s = "\"%s\",%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%d,%s,%s," % (
+        s = "MAKE_CMD(\"%s\",%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%d,%s,%s,%s,%s,%d)," % (
             self.name.lower(),
             get_optional_desc_string(self.desc, "summary"),
             get_optional_desc_string(self.desc, "complexity"),
             get_optional_desc_string(self.desc, "since"),
             _doc_flags_code(),
             get_optional_desc_string(self.desc, "replaced_by"),
             get_optional_desc_string(self.desc, "deprecated_since"),
+            "\"%s\"" % self.group,
```

Comment:
There is the group name ("generic") and a group enum value (COMMAND_GROUP_GENERIC). The existing command struct stores the enum value; the CLI code stores the string, so I added the string output here. Probably better to standardize the code on the same approach. Will fix.

Reply:
Turns out this is harder than I expected. The problem is `const char *COMMAND_GROUP_STR[]`, a constant array defined in server.c. To make the cli command struct consistent, the CLI would also need access to that array to retrieve the group name from the enum code. But it's currently only used in server.c and it's linked internally there. C doesn't let you forward declare a constant array, since the size is unknown. 

In short, it's much simpler to populate the group name as a string in the generated code than to propagate the enum to both. I guess the array of strings could be defined in commands.c and access to it wrapped in a function...

Reply:
We can add the array in commands.c , 
but also I think the strings values of the COMMAND command don't have to be the same, so arguably it's OK for redis-cli to define its own array too (as long as there's some static assertion to make sure it's the same size as the enum). 

But commands.c is probably better..

Reply:
Okay, I moved the array to `commands.c` with a wrapper function exposed in `commands.h`.